### PR TITLE
Fix units for MEKE advection

### PR DESCRIPTION
**Pull request by adcroft**
Three commits:
- Fix for units in the advection term for MEKE. Explanation in aac71f5.
- Removed some old commented-out code.
- Cleanup of comments and additional comments for clarification of units for some terms.

### CODE REVIEW

**Comment by adcroft** [comment]
Pipeline test at https://gitlab.gfdl.noaa.gov/ogrp/MOM6/pipelines/9449

File: src/parameterizations/lateral/MOM_MEKE.F90 [hunk 1]
```diff
@@ -266,7 +265,7 @@ subroutine step_forward_MEKE(MEKE, h, SN_u, SN_v, visc, dt, G, GV, US, CS, hu, h
       enddo ; enddo
       do i=is-1,ie+1
         I_mass(i,j) = 0.0
-        if (mass(i,j) > 0.0) I_mass(i,j) = 1.0 / mass(i,j)
+        if (mass(i,j) > 0.0) I_mass(i,j) = 1.0 / mass(i,j) ! [m2 kg-1]
```
Comment (Hallberg-NOAA) [inline src/parameterizations/lateral/MOM_MEKE.F90]: The correct units in the comment here should be '[R-1 Z-1 ~> m2 kg-1]'

File: src/parameterizations/lateral/MOM_MEKE.F90 [hunk 2]
```diff
@@ -401,15 +398,16 @@ subroutine step_forward_MEKE(MEKE, h, SN_u, SN_v, visc, dt, G, GV, US, CS, hu, h
       enddo ; enddo
       !$OMP parallel do default(shared) private(K4_here,Inv_K4_max)
       do J=js-1,je ; do i=is,ie
-        K4_here = CS%MEKE_K4
+        K4_here = CS%MEKE_K4 ! [L4 T-1 ~> m4 s-1]
         Inv_K4_max = 64.0 * sdt * ((G%dx_Cv(i,J)*G%IdyCv(i,J)) * max(G%IareaT(i,j), G%IareaT(i,j+1)))**2
         if (K4_here*Inv_K4_max > 0.3) K4_here = 0.3 / Inv_K4_max
 
+        ! Here the units of MEKE_vflux are [kg m-2 L4 T-3].
```
Comment (Hallberg-NOAA) [inline src/parameterizations/lateral/MOM_MEKE.F90]: The correct units here are '[R Z L4 T-3 ~> kg m2 s-3]'

File: src/parameterizations/lateral/MOM_MEKE.F90 [hunk 3]
```diff
@@ -457,12 +454,10 @@ subroutine step_forward_MEKE(MEKE, h, SN_u, SN_v, visc, dt, G, GV, US, CS, hu, h
             (MEKE%MEKE(i,j) - MEKE%MEKE(i,j+1))
       enddo ; enddo
       if (CS%MEKE_advection_factor>0.) then
-        !### I think that for dimensional consistency, this should be:
-        ! advFac = GV%H_to_RZ * CS%MEKE_advection_factor / sdt
-        advFac = US%kg_m3_to_R*GV%H_to_Z * CS%MEKE_advection_factor / dt
+        advFac = CS%MEKE_advection_factor / sdt ! [T-1]
```
Comment (Hallberg-NOAA) [inline src/parameterizations/lateral/MOM_MEKE.F90]: The units here should have the converted units, as '[T-1 ~> s-1]'

File: src/parameterizations/lateral/MOM_MEKE.F90 [hunk 4]
```diff
@@ -401,15 +398,16 @@ subroutine step_forward_MEKE(MEKE, h, SN_u, SN_v, visc, dt, G, GV, US, CS, hu, h
...
         Inv_K4_max = 64.0 * sdt * ((G%dx_Cv(i,J)*G%IdyCv(i,J)) * max(G%IareaT(i,j), G%IareaT(i,j+1)))**2
         if (K4_here*Inv_K4_max > 0.3) K4_here = 0.3 / Inv_K4_max
 
+        ! Here the units of MEKE_vflux are [kg m-2 L4 T-3].
         MEKE_vflux(i,J) = ((K4_here * (G%dx_Cv(i,J)*G%IdyCv(i,J))) * &
             ((2.0*mass(i,j)*mass(i,j+1)) / ((mass(i,j)+mass(i,j+1)) + mass_neglect)) ) * &
             (del2MEKE(i,j+1) - del2MEKE(i,j))
       enddo ; enddo
-      ! Store tendency arising from the bi-harmonic in del4MEKE
+      ! Store change in MEKE arising from the bi-harmonic in del4MEKE [L2 T-2].
```
Comment (Hallberg-NOAA) [inline src/parameterizations/lateral/MOM_MEKE.F90]: The units here should have the converted MKS units, as '[L2 T-2 ~> m2 s-2]'

**Comment by adcroft** [comment]
Yes - these are all because of a bad merge.

**Pull request merged**